Automattic/wp-calypso
# Theme Discovery: Back to Discover navigation.

Related to https://github.com/Automattic/wp-calypso/issues/82852

I've implemented a way to navigate back to the screen where we show the collection carousels when the user clicks on the `see all` button of any carousel.

This PR is only focused on the navigation functionality and not on the actual design of the `see all` screen. We can work on the design on a follow-up once we have a clear path forward. I did implement a basic UI that resembles the one found in our Figma designs.

![Captura de pantalla 2023-10-16 a las 14 45 13](https://github.com/Automattic/wp-calypso/assets/1989914/bf13479f-1861-45b6-84d2-e88e20e3f404)




## Proposed Changes

* Created a new view for tiers. This view excludes some elements to resemble the Figma designs (see PR description)
* Changed the themes controller so that it's aware of the collection view.
* We can toggle between collection view and legacy view on tiers and subjects. This will work but we currently only have a title and description for the 2 existing carousels (well, the description isn't final).

## Testing Instructions



* Open the Calypso Live Link and open the LoTS with `?flags=themes/discovery-lots`
* Click on a `See all` button.
* You should see a new screen that displays the Collection.
* The screen should have a back button.
* When clicking on the back button you should be redirected to the theme showcase URL without any filters or subjects applied.

Before:
https://github.com/Automattic/wp-calypso/assets/1989914/1cbc3a9b-d12e-4667-b008-165cfd40c34c

After:
https://github.com/Automattic/wp-calypso/assets/1989914/469fcae2-80b2-4ee9-8afb-9e53f1f8c94e


## Pre-merge Checklist
